langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiEmbeddingModel.java

```diff
@@ -102,7 +104,39 @@ public Response<List<Embedding>> embedAll(List<TextSegment> textSegments) {
                 .map(TextSegment::text)
                 .toList();
 
-        return embedTexts(texts);
+        List<List<String>> textBatches = splitList(texts, REQUEST_TEXT_MAX_ARRAY_SIZE);
+
+        return embedBatchedTexts(textBatches);
+    }
+
+    private List<List<String>> splitList(final List<String> inputList, final int size) {
```

Comment:
```suggestion
    private List<List<String>> partition(final List<String> inputList, final int size) {
```
nit

Reply:
I added a way to configure the segment range. The range is set to <1, 2048> based on OpenAI's documentation, which states: "Cannot be an empty string, and any array must be 2048 dimensions or less."
[OpenAI API Reference](https://platform.openai.com/docs/api-reference/embeddings/create)
And also I added missing test

Reply:
@Empatixx I am not sure we should restrict users to 2048 max. What happens if OpenAI lifts this requirement? What happens if users will use `OpenAiEmbeddingModel` to connect to other OpenAI-compatible embedding APIs that support higher number of parallel embeddings?
I guess checking that the number is greater than 0 is enough

Reply:
Understood, updated accordingly.